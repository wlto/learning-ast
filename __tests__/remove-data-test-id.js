const fs = require('fs');
const path = require('path');
const babel = require('@babel/core');
const plugin = require('../plugins/removeDataTestId');

const example = fs.readFileSync(
  path.resolve(__dirname, '../examples/remove-data-test-id.js'),
  'utf-8',
);

it('remove data-test-id attribute from JSX', () => {
  const {code} = babel.transform(example, {
    plugins: [plugin],
    presets: ['@babel/preset-react'],
  });
  expect(code).toMatchSnapshot();
});
