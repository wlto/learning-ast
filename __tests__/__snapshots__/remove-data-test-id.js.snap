// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`remove data-test-id attribute from JSX 1`] = `
"const React = require('react');

function App() {
  return /*#__PURE__*/React.createElement(\\"div\\", null, \\"world\\");
}"
`;
